lib/helpers.js

```diff
@@ -47,6 +57,72 @@ function fetchAndCache(request, options) {
   });
 }
 
+var staticIdb;
+function idbPromise() {
+  if (staticIdb) {
+    return Promise.resolve(staticIdb);
+  }
+
+  return simpleDB.open('sw-toolbox-lru-order').then(function(idb) {
+    staticIdb = idb;
+    return staticIdb;
+  });
+}
+
+var cacheExpirationPromiseChain;
+function queueCacheExpiration(request, cache, options) {
+  var cacheExpiration = cacheExpirationPromiseFactory.bind(null, request, cache, options);
+
+  if (cacheExpirationPromiseChain) {
+    cacheExpirationPromiseChain = cacheExpirationPromiseChain.then(cacheExpiration);
+  } else {
+    cacheExpirationPromiseChain = cacheExpiration();
+  }
+}
+
+function cacheExpirationPromiseFactory(request, cache, options) {
```

Comment:
Sigh. I had wanted to avoid having to use IndexedDB as, like, a DB with indexes and all, but I guess you're right about it being more efficient (though not simpler).

I'll take a look at making those modifications, though it might be abandoning the `SimpleDB` wrapper and going directly against the asynchronous API.